module/ColorJudge.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * @file ColorJudge.cpp
+ * @brief 色識別クラス
+ * @author hiroto0927, miyashita64
+ */
+
+#include "ColorJudge.h"
+
+ColorJudge::ColorJudge() {}
+
+// RGBをHSVに変形し、色を判別する
+COLOR ColorJudge::getColor(rgb_raw_t const& _rgb)
+{
+  int hue = 0;         // 色相(0-360)
+  int saturation = 0;  // 彩度(0-100)
+  int value = 0;       // 明度(0-255)
+
+  rgb_raw_t rgb;
+
+  // 環境光を打ち消す
+  rgb.r = (_rgb.r < MAX_RGB.r) ? _rgb.r * 256 / MAX_RGB.r : 256;
+  rgb.g = (_rgb.g < MAX_RGB.g) ? _rgb.g * 256 / MAX_RGB.g : 256;
+  rgb.b = (_rgb.b < MAX_RGB.b) ? _rgb.b * 256 / MAX_RGB.b : 256;
+
+  // RGBの中の最大・最小値を求める
+  int max = (rgb.r > rgb.g && rgb.r > rgb.b) ? rgb.r : (rgb.g > rgb.b) ? rgb.g : rgb.b;
+  int min = (rgb.r < rgb.g && rgb.r < rgb.b) ? rgb.r : (rgb.g < rgb.b) ? rgb.g : rgb.b;
```

Comment:
個人的にはやっぱりこっちの方がわかりやすいです（短いので）
処理は変わらないのでお任せします。
```suggestion
  #include <algorithm>
  int max = std::max({rgb.r, rgb.g, rgb.b});
  int max = std::max({rgb.r, rgb.g, rgb.b});
```